src/WebJobs.Script/StorageProvider/IAzureStorageProvider.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Azure.Storage.Blobs;
+
+namespace Microsoft.Azure.WebJobs.Script
+{
+    public interface IAzureStorageProvider
+    {
+        bool TryGetBlobServiceClientFromConnectionString(out BlobServiceClient client, string connectionString);
```

Comment:
The fact that both of these methods take a string param while the only difference in the method name is "string" makes potential for subtle bugs where the wrong method is called. Did you consider introducing some sort of wrapper type for the connection scenario, just to help make the semantic difference between these methods more pronounced/clear?

Reply:
Personally I think the verboseness of the method is enough to distinguish between using a connection vs. connection string. Adding a wrapper to hold a single property may be adding too much complexity. Current tests that rely on AzureStorageProvider would fail if one is used over the other.